src/main/java/com/epam/spring/homework2/other/CustomBeanPostProcessor.java

```diff
@@ -0,0 +1,16 @@
+package com.epam.spring.homework2.other;
+
+import com.epam.spring.homework2.validation.BeanValidator;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+
+public class CustomBeanPostProcessor implements BeanPostProcessor {
+
+    @Override
+    public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
```

Comment:
Hi Viktor. To be honest I like your first implementation of this task (in the previous commit) more than this one )))
1. In the previous implementation all logic related to validation was concentrated in one place. In this variant validation logic is spread through all beans using **validation** method
2. We have code duplication. The same code is used multiple time.
3. Also in some way we violates Single Responsibility Principle placing validation logic in the class that is supposed to use just for storing data.
Please try to come up with some other approach or maybe just come back to the previous implementation that is good )))

Thank you in advance!

Reply:
Hi Roman. I come back to the previous implementation but I added annotation for filtering beans that have to be validated. Thank you for advice.

Reply:
Hi Viktor! Thank you for your efforts and time! It looks good!
We can see that some logic can be moved to the separate methods to simplify code and avoid duplication.
Please try to refactor your code using these methods: 

1. for retrieving field by name & type and :
```

    private <T> Optional<Field> getFieldByNameAndType(Object bean, String fieldName, Class<T> clazz) {
        Optional<Field> fieldOptional;

        try {
            Field field = bean.getClass().getDeclaredField(fieldName);
            if (field.getType().equals(clazz)) {
                fieldOptional = Optional.of(field);
            } else {
                fieldOptional = Optional.empty();
            }
        } catch (NoSuchFieldException e) {
            fieldOptional = Optional.empty();
        }

        return fieldOptional;
    }
```

2. Retrieving value from field
```
    private <T> Optional<T> getValue(Object bean, Field field) {
        try {
            field.setAccessible(true);
            return Optional.of((T)field.get(bean));
        } catch (IllegalAccessException | NullPointerException e) {
            return Optional.empty();
        }
    }
```

Feel free to contact me in the Teams in case you have any questions

Reply:
Hi Viktor! Great job! Thanks for implementing these changes in the declarative style.
Let`s perform final change. Please use this snippet of code as an example and perform corresponding changes for 'value' part of the code. We just want to make code in more functional style))):
```
            getFieldByNameAndType(bean, "name", String.class)
                    .map(field -> getValue(bean, field))
                    .filter(v -> !v.isPresent())
                    .ifPresent(field ->
                        System.out.println(beanName + ". 'name' can't be null")
                    );
```
Also let simplify code a bit and show messages only for broken violation checks. So in this case **list** variant become redundant.
I approve PR, so once you finish just merge this branch into `main` branch
Thank you very much for you work!